### PR TITLE
PLT-329 correctly support command line data types

### LINKED ISSUE #90: bug: strings are converted into int
Do an API call that sends a phone number as input, wick converts that to an integer.

We need a way to tell the CLI the data type of the input value

### PR DESCRIPTION
**Pull request by muzzammilshahid**
### Description and Context
It is expected that this PR is in a branch with a JIRA ticket PLT-329 which thoroughly describes the change. fixes #90 and fixed golint errors and tests

### What kind of change does this PR introduce?

Put an `x` in all the boxes that apply:

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

### Checklist:

Go over all the following points, and put an `x` in all the boxes that apply.

If you're unsure about any of these, don't hesitate to ask teammates. We're here to help!

- [ ] My code follows the code style of this project.
- [ ] I have added tests to cover my changes.
- [ ] This changes or adds a WAMP RPC/PubSub
    - [ ] I have updated the WAMP API Catalogue in Notion and/or added the new schema to the st-schemas repo
- [ ] This changes logic or functionality
    - [ ] I have updated the README.md in this repo.

### CODE REVIEW

File: core/helpers_test.go [hunk 1]
```diff
@@ -140,71 +140,31 @@ func TestUrlSanitization(t *testing.T) {
...
-		t.Error("error in list conversion")
-	}
-
-	if _, canConvert := wampList[2].(float64); canConvert == false {
-		t.Error("error in list conversion")
-	}
+	expectedList := wamp.List{"string", 1, 1.1, true, []interface{}{"group_1", "group_2", 1.1, true},
+		map[string]interface{}{"firstKey": "value", "secondKey": 2.1},
+		[]map[string]interface{}{{"firstKey": "value", "secondKey": 2.1}, {"firstKey": "value", "secondKey": 2.1}},
+		"123", "123", "true"}
```
Comment (bboozzoo) [inline core/helpers_test.go:151]:
```suggestion
	expectedList := wamp.List{
		"string", 1, 1.1, true, 
		"123", "123", "true",
		// converted from JSON
		[]interface{}{"group_1", "group_2", 1.1, true},
		map[string]interface{}{"firstKey": "value", "secondKey": 2.1},
		[]map[string]interface{}{
			{"firstKey": "value", "secondKey": 2.1}, 
			{"firstKey": "value", "secondKey": 2.1},
		},
	}
```

File: core/helpers_test.go [hunk 2]
```diff
@@ -140,71 +140,31 @@ func TestUrlSanitization(t *testing.T) {
...
-		t.Error("error in map conversion")
-	}
-
-	if _, canConvert := wampDict["list"].([]interface{}); canConvert == false {
-		t.Error("error in map conversion")
-	}
+	expectedDict := wamp.Dict{"string": "string", "int": 1, "float": 1.1, "bool": true,
+		"list": []interface{}{"group_1", "group_2", 1.1, true}, "json": map[string]interface{}{"firstKey": "value", "secondKey": 2.2},
+		"jsonList":     []map[string]interface{}{{"firstKey": "value", "secondKey": 2.2}, {"firstKey": "value", "secondKey": 2.2}},
+		"stringNumber": `"123`, "stringFloat": "1.23", "stringBool": "true"}
```
Comment (bboozzoo) [inline core/helpers_test.go:166]:
```suggestion
	expectedDict := wamp.Dict{
		"string": "string", "int": 1, "float": 1.1, "bool": true,
		"stringNumber": `"123`, "stringFloat": "1.23", "stringBool": "true",
		"list": []interface{}{
			"group_1", "group_2", 1.1, true,
		},
		"json": map[string]interface{}{"firstKey": "value", "secondKey": 2.2},
		"jsonList":     []map[string]interface{}{
			{"firstKey": "value", "secondKey": 2.2}, 
			{"firstKey": "value", "secondKey": 2.2},
		},
	}
```

File: core/helpers_test.go [hunk 3]
```diff
@@ -140,71 +140,31 @@ func TestUrlSanitization(t *testing.T) {
...
-	if _, canConvert := wampDict["int"].(int); canConvert == false {
-		t.Error("error in map conversion")
-	}
-
-	if _, canConvert := wampDict["float"].(float64); canConvert == false {
-		t.Error("error in map conversion")
-	}
+		"list": `["group_1","group_2", 1.1, true]`, "json": `{"firstKey":"value", "secondKey":2.2}`,
+		"jsonList":     `[{"firstKey":"value", "secondKey":2.2}, {"firstKey":"value", "secondKey":2.2}]`,
+		"stringNumber": `""123"`, "stringFloat": `'1.23'`, "stringBool": `"true"`}
```
Comment (bboozzoo) [inline core/helpers_test.go:161]:
```suggestion
	inputDict := map[string]string{
		"string": "string", "int": "1", "float": "1.1", "bool": "true",
		"stringNumber": `""123"`, "stringFloat": `'1.23'`, "stringBool": `"true"`,
		"list": `["group_1","group_2", 1.1, true]`,
		"json": `{"firstKey":"value", "secondKey":2.2}`,
		"jsonList": `[{"firstKey":"value", "secondKey":2.2}, {"firstKey":"value", "secondKey":2.2}]`,
	}
```

File: core/helpers.go [hunk 4]
```diff
@@ -62,12 +62,15 @@ func listToWampList(args []string) wamp.List {
 	}
 
 	for _, value := range args {
-
 		var mapJson map[string]interface{}
 		var mapList []map[string]interface{}
 		var simpleList []interface{}
 
-		if number, errNumber := strconv.Atoi(value); errNumber == nil {
+		if strings.HasPrefix(value, `'`) && strings.HasSuffix(value, `'`) {
```
Comment (bboozzoo) [inline core/helpers.go:69]: Also, is there some sort of `--help` output that should be updated to describe the formatting? The output of `wick help call` or `wick help publish` isn't exactly informative.

[the remark this answers is not in the document]
Comment (muzzammilshahid) [reply core/helpers.go]: Done Thanks.

File: core/helpers.go [hunk 5]
```diff
@@ -91,12 +94,15 @@ func dictToWampDict(kwargs map[string]string) wamp.Dict {
 	var keywordArguments wamp.Dict = make(map[string]interface{})
 
 	for key, value := range kwargs {
-
 		var mapJson map[string]interface{}
 		var mapList []map[string]interface{}
 		var simpleList []interface{}
 
-		if number, errNumber := strconv.Atoi(value); errNumber == nil {
+		if strings.HasPrefix(value, `'`) && strings.HasSuffix(value, `'`) {
```
[the remark this answers is not in the document]
Comment (muzzammilshahid) [reply core/helpers.go]: Done

File: core/helpers_test.go (hunk 1 above)
Comment (muzzammilshahid) [reply core/helpers_test.go]: Done

File: core/helpers_test.go (hunk 2 above)
Comment (muzzammilshahid) [reply core/helpers_test.go]: Done

**Review by bboozzoo — APPROVED** [review]
LGTM

**Pull request merged**